src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -106,9 +106,18 @@ protected function extractAttributes($object, $format = null, array $context = [
         $checkPropertyInitialization = \PHP_VERSION_ID >= 70400;
 
         // properties
-        foreach ($reflClass->getProperties(\ReflectionProperty::IS_PUBLIC) as $reflProperty) {
-            if ($checkPropertyInitialization && !$reflProperty->isInitialized($object)) {
-                continue;
+        foreach ($reflClass->getProperties() as $reflProperty) {
+            if ($checkPropertyInitialization) {
+                if (!$reflProperty->isPublic()) {
```

Comment:
Maybe could you store the result of this call in a variable to reuse it in the last check?